File: 0008-string-to-integer-atoi/0008-string-to-integer-atoi.py
class Solution:
    def myAtoi(self, s: str) -> int:

        s=s.lstrip()
        if not s:
            return 0
        sign=1
        a=0

        if s[0]=='-':
            sign=-1
        for i in range(len(s)):
            if i==0 and (s[i]=='+' or s[i]=='-'):
                continue
            elif s[i].isdigit():
                a=a*10+int(s[i])
            else:
                break
        if -2**31<=a*sign<=2**31-1:
            return a*sign 
        elif(sign==-1):
            return -2**31
        else:
            return 2**31-1
        

File: 0008-string-to-integer-atoi/0008-string-to-integer-atoi.cpp
class Solution {
public:
    int myAtoi(string s) {
        int sign=1;
        long long ans=0;

        int idx = 0;
        while (idx < s.size() && s[idx] == ' ') idx++;
        s = s.substr(idx);

        if (s[0]=='-') {
            sign=-1;
        }

        for(int i=0;i<s.size();i++){
            if(i==0 && (s[0]=='-' || s[0]=='+')){
                continue;
            }
            else if(isdigit(s[i])){
                ans = ans*10 + (s[i]-'0');

                if (sign == 1 && ans > INT_MAX) return INT_MAX;
                if (sign == -1 && -ans < INT_MIN) return INT_MIN;
            }
            else{
                break;
            }
        }

        return (int)(ans * sign);
    }
};
